src/main/java/net/glowstone/entity/passive/GlowAbstractHorse.java

```diff
@@ -67,7 +79,7 @@ private int getHorseFlags() {
         }
         if (this instanceof GlowHorse) {
             GlowHorse horse = (GlowHorse) this;
-            if (getInventory() != null && ((HorseInventory) getInventory()).getSaddle() != null) {
+            if (getInventory() != null && InventoryUtil.isEmpty(getInventory().getSaddle())) {
```

Comment:
This should be the opposite (`!InventoryUtil.isEmpty`)

Reply:
oops, my bad